src/execCmd.ts

```diff
@@ -303,3 +321,29 @@ export function execCmd<T = Collection>(
     }
   }
 }
+
+/**
+ * Determines the execCmd timeout value using the following logic:
+ *  1. Examine the command to be executed and determine if it contains a `--wait` param.
+ *  2. If `--wait` param is present, use the value as the timeout.
+ *  3. If `--wait` param is not present, use the value from the `callTimeout` param.
+ *  4. Neither `--wait` param nor `callTimeout` param can be negative.
+ *
+ * @param cmd
+ * @param callTimeout
+ */
+export function getWaitTime(cmd: string, callTimeout = Duration.minutes(DEFAULT_WAIT_TIME).milliseconds): number {
+  const waitTimeRegex = /(?:(-w=?\s*?)|(--wait=?\s*?))([-+]?\d+)/;
+  const waitTimeText = cmd.match(waitTimeRegex)?.[3];
+  if (waitTimeText) {
+    const waitMilliseconds = Duration.minutes(Number(waitTimeText || '0')).milliseconds;
+    if (waitMilliseconds < 0) {
+      throw messages.createError('waitTimeNegative', [waitTimeText]);
+    }
+    return waitMilliseconds;
+  }
+  if (callTimeout < 0) {
```

Comment:
mdapi commands use `--wait -1` to mean don't stop waiting.  This would break those.